base/mpfr.jl

```diff
@@ -687,9 +687,11 @@ for (fJ, fC, fI) in ((:+, :add, 0), (:*, :mul, 1))
 end
 
 function -(x::BigFloat)
-    z = BigFloat()
-    ccall((:mpfr_neg, libmpfr), Int32, (Ref{BigFloat}, Ref{BigFloat}, MPFRRoundingMode), z, x, ROUNDING_MODE[])
-    return z
+    setprecision(precision(x)) do
+        z = BigFloat()
```

Comment:
Why not simply `z = BigFloat(;precision=precision(x))` ? That would avoid changing the global setting.

Reply:
No reason. `z = BigFloat(;precision=precision(x))` is more readable, but I hadn't thought of it at the time. But I don't think `setprecision` will change anything outside the `do` block.

Reply:
It is not thread-safe, though. That means during task switches during the `do` block, the global setting of one of the tasks could be undefined.